share/eds/store.go

```diff
@@ -565,7 +565,7 @@ type inMemoryOnceMount struct {
 	buf *bytes.Buffer
 
 	readOnce atomic.Bool
-	mount.Mount
```

Comment:
The issue makes sense. When dagstore instantiates the registered mount it should know the exact type. It could derive the type from given `&mount.FileMount{}` but it's a bit advanced, thus the issue on DAGStore.